### PR TITLE
Add common and crypto function

### PR DESCRIPTION
**Pull request by lucusfly**

### CODE REVIEW

File: common/hash.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	var a Hash
+	a.SetBytes(b)
+	return a
+}
+
+func StringToAddress(s string) Hash {
+	return BytesToAddress([]byte(s))
+}
+
+// Sets the hash to the value of b. If b is larger than len(a) it will panic
```
Comment (WTer) [inline common/hash.go]: comments should be started with SetBytes

File: common/hash.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (WTer) [inline common/hash.go]: nice to add unittests for these files.

File: common/hash.go (hunk 1 above)
Comment (lucusfly) [reply common/hash.go]: resolved

File: common/hash.go (hunk 2 above)
Comment (lucusfly) [reply common/hash.go]: add test

**Pull request merged**